src/openfunction/runtime.ts

```diff
@@ -243,4 +261,216 @@ class DaprRuntime extends OpenFunctionRuntime {
 
     return Promise.allSettled(actions);
   }
+
+  /**
+   * Save the data to the state store
+   * @param data The data for save operation
+   * @param db The state store to save the data
+   * @returns The promise of the save action being executed.
+   */
+  #save(data: object, db?: string): Promise<object> {
+    if (!this.context.states) {
+      return Promise.resolve([]);
+    }
+    const states = this.context.states;
+    const actions = Object.keys(states)
+      .filter(v => !db || v === db)
+      .map((storeName: string) => {
+        const component: ComponentSpec = states[storeName];
+        if (ContextUtils.IsStateComponent(component)) {
+          return this.daprClient.state.save(
+            storeName,
+            data as KeyValuePairType[]
+          );
+        }
+        return Promise.resolve(undefined);
+      })
+      .values();
+    return Promise.allSettled(actions);
```

Comment:
[blocker] `Promise.allSettled` is used for waiting for multiple promises, it's not suitable for current single invocation, just return simple values, again this code reference may help: https://github.com/dapr/js-sdk/blob/main/src/implementation/Client/HTTPClient/state.ts